ZenPacks/zenoss/OpenStackInfrastructure/apiclients/session.py

```diff
@@ -330,16 +332,20 @@ def _detect_api_version(self):
             try:
                 data = json.loads(response_body)
                 version_url = {}
+                best_version = "v0.0"
                 for version in data['versions']['values']:
                     if version['status'] != 'stable':
                         continue
+                    if LooseVersion(version['id']) < LooseVersion(best_version):
+                        continue
+                    if LooseVersion(version['id']) >= LooseVersion("v4.0"):
+                        continue
+                    best_version = version['id']
```

Comment:
I appreciate the rigorous implementation, but since other places in the code are just using "startswith("v3""), i think we should stick with that for the time being- this is almost certainly more flexible but I want to minimize changes at this point in the cycle.